### PR TITLE
Onboarding: Convert tax forms to optional step

### LINKED ISSUE #1569: Disable mandatory tax upload in the onboarding app (Onboarding)
In the onboarding app, the issuer can decide if tax document upload is required or not. When activating onboarding default setting is that it is required.

### PR DESCRIPTION
**Pull request by sophialittlejohn**
### Description

This pull request...
- makes tax info optional and includes it in the sign sub doc step if required by the issuer
- provides a checkbox for issuers to make tax docs mandatory or not. Default is set to mandatory as soon as onboarding is enabled.

Closes #1569

### Approvals

- [x] Dev
- [ ] Dev
- [ ] Designer
- [ ] Product

### Screenshots

Screenshot 2023-09-08 at 16 40 29
Screenshot 2023-09-08 at 16 41 07

### CODE REVIEW

File: centrifuge-app/src/components/Dialogs/EditOnboardingEmailAddressDialog.tsx [hunk 1]
```diff
@@ -49,27 +49,27 @@ export const EditOnboardingEmailAddressDialog = ({ isDialogOpen, setIsDialogOpen
...
-              loading={isLoading}
-              disabled={isLoading || !isValid}
-              loadingMessage="Updating"
-            >
-              Update
-            </Button>
-          </Shelf>
-        </Stack>
-      </Box>
+      {/* <Box p={4}> */}
```
Comment (jpangelle) [inline centrifuge-app/src/components/Dialogs/EditOnboardingEmailAddressDialog.tsx:54]: could just remove?

File: centrifuge-app/src/utils/getActiveOnboardingStep.ts [hunk 2]
```diff
@@ -6,41 +6,38 @@ const BASE_ENTITY_STEPS = {
   VERIFY_BUSINESS: 3,
```
Comment (jpangelle) [inline centrifuge-app/src/utils/getActiveOnboardingStep.ts:6]: If the user is on the tax upload step and then you refresh the page, does the app properly jump to the tax upload step?

Comment (sophialittlejohn) [reply centrifuge-app/src/utils/getActiveOnboardingStep.ts:6]: now the tax step is included in the sign document step so no more official step for taxes. I just checked that refreshing the page puts you on the correct step

Comment (jpangelle) [reply centrifuge-app/src/utils/getActiveOnboardingStep.ts:6]: ah ok cool

**Review by jpangelle — APPROVED** [review]

**Pull request merged**